File: base/src/main/java/com/lotteryviewer/base/util/SafetyHandler.java
package com.lotteryviewer.base.util;

import android.os.Handler;
import android.os.Message;

import java.lang.ref.WeakReference;

/**
 * @Author: duke
 * @DateTime: 2021-03-17 12:42:42
 * @Description: 功能描述：
 */
public class SafetyHandler  extends Handler {
    private final WeakReference<Delegate> mWeakReference;


    private SafetyHandler() {
        this(null);
    }

    private SafetyHandler(SafetyHandler.Delegate delegate) {
        mWeakReference = new WeakReference<>(delegate);
    }

    public static SafetyHandler create(SafetyHandler.Delegate delegate) {
        return new SafetyHandler(delegate);
    }

    @Override
    public void handleMessage(Message msg) {
        super.handleMessage(msg);
        if (mWeakReference == null || mWeakReference.get() == null) {
            return;
        }

        mWeakReference.get().onReceivedHandlerMessage(msg);
    }

    public interface Delegate {
        void onReceivedHandlerMessage(Message message);
    }
}

File: base/src/main/java/com/lotteryviewer/base/util/SafetyHandler.kt
package com.lotteryviewer.base.util

import android.os.Handler
import android.os.Looper
import android.os.Message
import java.lang.ref.WeakReference

/**
 * @Author: duke
 * @DateTime: 2021-03-17 12:42:42
 * @Description: 功能描述：
 */
class SafetyHandler private constructor(delegate: Delegate? = null) :
    Handler(Looper.getMainLooper()) {

    companion object {
        fun create(delegate: Delegate?): SafetyHandler {
            return SafetyHandler(delegate)
        }
    }

    private val mWeakReference: WeakReference<Delegate?>?

    init {
        mWeakReference = WeakReference(delegate)
    }

    override fun handleMessage(msg: Message) {
        super.handleMessage(msg)
        mWeakReference?.get()?.onReceivedHandlerMessage(msg)
    }

    interface Delegate {
        fun onReceivedHandlerMessage(message: Message?)
    }

}